src/scripts/main.js

```diff
@@ -1,3 +1,324 @@
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    for (let j = 1; j < board[i].length; j++) {
+      if (board[i][j - 1] === board[i][j]) {
+        board[i][j - 1] = board[i][j - 1] * 2;
+        board[i][j] = 0;
+        score += board[i][j - 1];
+      }
+    }
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]);
+  }
+
+  newCell();
+  display();
+  ifWin();
+}
+
+function moveUp() {
+  for (let i = 0; i < board.length; i++) {
+    let newColumn = [];
+
+    for (let j = 0; j < board[0].length; j++) {
+      newColumn.push(board[j][i]);
+    }
+
+    newColumn = shiftRowLeft(newColumn);
+
+    for (let j = 1; j < newColumn.length; j++) {
+      if (newColumn[j - 1] === newColumn[j]) {
+        newColumn[j - 1] = newColumn[j - 1] * 2;
+        newColumn[j] = 0;
+        score += newColumn[j - 1];
+      }
+    }
+
+    newColumn = shiftRowLeft(newColumn);
+
+    for (let j = 0; j < newColumn.length; j++) {
+      board[j][i] = newColumn[j];
+    }
+  }
+
+  newCell();
+  display();
+  ifWin();
+}
+
+function moveDown() {
+  for (let i = 0; i < board.length; i++) {
+    let newColumn = [];
+
+    for (let j = 0; j < board[0].length; j++) {
+      newColumn.push(board[j][i]);
+    }
+
+    newColumn = shiftRowLeft(newColumn).reverse();
+
+    for (let j = 1; j < newColumn.length; j++) {
+      if (newColumn[j - 1] === newColumn[j]) {
+        newColumn[j - 1] = 0;
+        newColumn[j] = newColumn[j] * 2;
+        score += newColumn[j];
+      }
+    }
+
+    newColumn = shiftRowLeft(newColumn).reverse();
+
+    for (let j = 0; j < newColumn.length; j++) {
+      board[j][i] = newColumn[j];
+    }
+  }
+
+  newCell();
+  display();
+  ifWin();
+}
+
+function changeAdditionalClassCell(element, newAddClass) {
+  element.classList = '';
+  element.classList.add('field-cell');
+
+  if (newAddClass) {
+    element.classList.add(newAddClass);
+  }
+}
+
+function ifWin() {
+  for (const row of board) {
+    for (const cell of row) {
+      if (cell === 2048) {
+        messageWin.classList.remove('hidden');
+        start.textContent = 'Start';
+        start.classList.remove('restart');
+        start.classList.add('start');
+        started = false;
+        messageStart.classList.remove('hidden');
+      }
+    }
+  }
+}
+
+function freeCell() {
+  for (const row of board) {
+    for (const cell of row) {
+      if (cell === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function ifLose() {
+  let lose = true;
+  const temp = [...board];
+  const tempProbabilityCount = probabilityCount;
+  const tempScore = score;
+
+  moveLeft();
+
+  if (freeCell()) {
+    lose = false;
+    board = [...temp];
+    probabilityCount = tempProbabilityCount;
+    score = tempScore;
+
+    return lose;
+  }
+
+  moveUp();
+
+  if (freeCell()) {
+    lose = false;
+    board = [...temp];
+    probabilityCount = tempProbabilityCount;
+    score = tempScore;
+
+    return lose;
+  }
```

Comment:
You do the same actions here in `if` statements. Would be good to create a separate function for it to not repeat the same each time